### PR TITLE
20971 add notes about relaxed range restrictions on `formatRange` and `selectRange`

### LINKED ISSUE #20971: Firefox 105 devnotes: some other things worth covering in the notes
### MDN URL

https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Releases/105

### What specific section or headline is this issue about?

_No response_

### What information was incorrect, unhelpful, or incomplete?

Nothing is incorrect, unhelpful. It's great. Thanks for keeping the devnotes.

### What did you expect to see?

I read the end-user release notes, and found that there are things that are developer related but aren't covered in the devnotes:

> Support for the Offscreen Canvas DOM API with full context and font support. The [OffscreenCanvas API](https://developer.mozilla.org/docs/Web/API/OffscreenCanvas) provides a canvas that can be rendered off-screen in both Window and Web Worker contexts.

> Firefox is now compliant with the User Timing L3 specification, which adds additional optional arguments to the `performance.mark` and `performance.measure` methods to provide custom start times, end times, duration, and attached details.

You might want to mention [1780545 - Remove range restrictions from Intl.DateTimeFormat, Intl.NumberFormat, and Intl.PluralRules](https://bugzilla.mozilla.org/show_bug.cgi?id=1780545) in the JavaScript section. It's a sort of breaking change (but for good), and Safari covers it in their [TP 151](https://webkit.org/blog/13093/release-notes-for-safari-technology-preview-151/). 

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/mozilla/firefox/releases/105`
* MDN URL: https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Releases/105
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/mozilla/firefox/releases/105/index.md
* Last commit: https://github.com/mdn/content/commit/331d598bb92fcddcf8f194dd8a62239b4a29ffdf
* Document last modified: 2022-09-20T15:50:51.000Z

</details>

### PR DESCRIPTION
**Pull request by bsmth**
This change adds a note about relaxed ordering rules for

* `Intl.DateTimeFormat`
* `Intl.NumberFormat`
* `Intl.PluralRules`

It's now possible to compare values such as negative date ranges

```javascript
// Given startDate of 'Jan 20th' and an endDate of 'Jan 10th'
const date1 = new Date(Date.UTC(2022, 0, 20, 10, 0, 0));
const date2 = new Date(Date.UTC(2022, 0, 10, 10, 0, 0));

const fmt2 = new Intl.DateTimeFormat("en", {
  year: 'numeric',
  month: 'short',
  day: 'numeric',
});

console.log(fmt2.formatRange(date2, date1));
// > "Jan 10 – 20, 2022"
```

## Bugs

https://bugzilla.mozilla.org/show_bug.cgi?id=1780545

Fixes #20971

### CODE REVIEW

File: files/en-us/mozilla/firefox/releases/105/index.md [hunk 1]
```diff
@@ -25,14 +25,19 @@ No notable changes.
 
 ### JavaScript
 
-No notable changes.
+- Range restrictions are relaxed on `formatRange` and `selectRange` functions for [`Intl.DateTimeFormat`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat), [`Intl.NumberFormat`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat), and [`Intl.PluralRules`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/PluralRules).
+  For example, `formatRange(x, y)` can be called where and `x` > `y` such as `formatRange(-500, -1000)`.
+  The intention of this change is to relax ordering rules so that negative ranges (e.g., "losses ranging from -$500 to -$1000") can be formatted correctly ({{bug(1780545)}}).
```
Comment (dipikabh) [inline files/en-us/mozilla/firefox/releases/105/index.md:30]: I think the examples and specific details can be covered in the `formatRange` and `selectRange` pages (probably separate PR). 

```suggestion
- Range restrictions have been relaxed on `formatRange` and `selectRange` functions for [`Intl.DateTimeFormat`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat), [`Intl.NumberFormat`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat), and [`Intl.PluralRules`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/PluralRules) objects. This change now allows negative ranges ({{bug(1780545)}}).
```

File: files/en-us/mozilla/firefox/releases/105/index.md [hunk 2]
```diff
@@ -25,14 +25,19 @@ No notable changes.
...
 
 ### APIs
 
 #### DOM
 
 - The [TextDecoderStream](/en-US/docs/Web/API/TextDecoderStream) and [TextEncoderStream](/en-US/docs/Web/API/TextEncoderStream) interfaces, part of the [Encoding API](/en-US/docs/Web/API/Encoding_API), are now supported ({{bug(1486949)}}).
 
+- The [OffscreenCanvas API](/en-US/docs/Web/API/OffscreenCanvas) provides a canvas that can be rendered off-screen in both Window and [Web Worker contexts](/en-US/docs/Web/API/OffscreenCanvas#asynchronous_display_of_frames_produced_by_an_offscreencanvas), decoupling the DOM and the Canvas API so that `<canvas>` elements are no longer dependent on the DOM.
+  The [OffscreenCanvasRenderingContext2D](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) interface is supported and enabled by default ({{bug(1746110)}}).
```
Comment (dipikabh) [inline files/en-us/mozilla/firefox/releases/105/index.md:39]:
```suggestion
- The [OffscreenCanvas](/en-US/docs/Web/API/OffscreenCanvas) API now provides a canvas that can be rendered off-screen in both window and [web worker](/en-US/docs/Web/API/OffscreenCanvas#asynchronous_display_of_frames_produced_by_an_offscreencanvas) contexts, decoupling the DOM and the Canvas API. This allows `<canvas>` elements to no longer be dependent on the DOM.
- The [OffscreenCanvasRenderingContext2D](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) interface is now supported and enabled by default ({{bug(1746110)}}).
```
- Please check if the lower casing suggestions are correct
- Added suggestion to split the long sentence into two
- Should 'OffscreenCanvasRenderingContext2D' be a separate point?

**Review by dipikabh — CHANGES_REQUESTED** [review]
Hi @bsmth, I've added a few suggestions for your consideration.

File: files/en-us/mozilla/firefox/releases/105/index.md (hunk 2 above)
Comment (bsmth) [reply files/en-us/mozilla/firefox/releases/105/index.md:39]: Casing suggestions look good. I'd lean towards keeping it in one point as the `OffscreenCanvasRenderingContext2D` is what made up most of the implementation work for this as a whole. What do you think?

File: files/en-us/mozilla/firefox/releases/105/index.md (hunk 1 above)
Comment (bsmth) [reply files/en-us/mozilla/firefox/releases/105/index.md]: Sounds good to me

File: files/en-us/mozilla/firefox/releases/105/index.md (hunk 2 above)
Comment (bsmth) [reply files/en-us/mozilla/firefox/releases/105/index.md]: Added improvements in https://github.com/mdn/content/pull/21011/commits/99794438270ea412860e1cff5c1eb7ffd9d15937

**Review by dipikabh — APPROVED** [review]
Thank you! 👍

**Pull request merged**